pkg/base1/test-file.js

```diff
@@ -1,249 +1,129 @@
+QUnit.test("read non-regular", async assert => {
+    try {
+        await cockpit.file(dir, { binary: true }).read();
+        assert.ok(false, "should have failed");
+    } catch (error) {
+        assert.equal(error.problem, "internal-error", "got error");
+    }
 });
 
-QUnit.test("read large", function (assert) {
-    const done = assert.async();
-    assert.expect(2);
-    cockpit.file(dir + "/large.bin", { binary: true }).read()
-            .done(function(resp) {
-                assert.equal(resp.length, 512 * 1024, "correct result");
-            })
-            .always(function() {
-                assert.equal(this.state(), "resolved", "didn't fail");
-                done();
-            });
+QUnit.test("read large", async assert => {
+    const resp = await cockpit.file(dir + "/large.bin", { binary: true }).read();
+    assert.equal(resp.length, 512 * 1024, "correct result");
 });
 
-QUnit.test("read too large", function (assert) {
-    const done = assert.async();
-    assert.expect(2);
-    cockpit.file(dir + "/large.bin", { binary: true, max_read_size: 8 * 1024 }).read()
-            .fail(function(error) {
-                assert.equal(error.problem, "too-large", "got error");
-            })
-            .always(function() {
-                assert.equal(this.state(), "rejected", "failed");
-                done();
-            });
+QUnit.test("read too large", async assert => {
+    try {
+        await cockpit.file(dir + "/large.bin", { binary: true, max_read_size: 8 * 1024 }).read();
+        assert.ok(false, "should have failed");
+    } catch (error) {
+        assert.equal(error.problem, "too-large", "got error");
+    }
 });
 
 /* regression: passing 'binary: false' made cockpit-ws close the whole connection */
-QUnit.test("binary false", function (assert) {
-    const done = assert.async();
-    assert.expect(1);
-    cockpit.file(dir + "/foo.bin", { binary: false }).read()
-            .done(function() {
-                assert.equal(this.state(), "resolved", "didn't fail");
-                done();
-            });
+QUnit.test("binary false", async assert => {
+    await cockpit.file(dir + "/foo.bin", { binary: false }).read();
+    assert.ok(true, "did not crash");
 });
 
-QUnit.only("simple replace", function (assert) {
-    const done = assert.async();
-    assert.expect(3);
-    cockpit.file(dir + "/bar").replace("4321\n")
-            .always(function() {
-                assert.equal(this.state(), "resolved", "didn't fail");
-                cockpit.spawn(["cat", dir + "/bar"])
-                        .done(function (res) {
-                            assert.equal(res, "4321\n", "correct content");
-                            cockpit.spawn(["stat", "-c", "%a", dir + "/bar"])
-                                    .done(res => {
-                                        assert.equal(res, "644\n", "correct permissions");
-                                        done();
-                                    });
```

Comment:
Argh, this was a local test addition which I accidentally committed in PR #18193. It's  not super-wrong, but it would fail in environments with tight umask. I reverted this again. That's what caused the conflict.